### PR TITLE
Make Forcings Engine Installable via pip

### PR DESCRIPTION
**Pull request by program--**
Hi Jason! This PR adds a pyproject.toml so that the forcings engine is installable via `pip`.

In summary:
- Adds `pyproject.toml` to root directory.
- Adds `__init__.py` to `core/` to initialize it as a submodule.
- Adds `__version__` entry into package init.
- Adds `build/` directory to `.gitignore`.
- Adds top-level BMI class alias, `BMIForcingsEngine`.
- Uses relative imports for intra-package imports.
- Replaces deprecated numpy aliases with base types (`np.int` and `np.bool`).
- Adds try/except around ESMF imports to handle the name change from `ESMF` to `esmpy`

I also think we can move this into the NOAA-OWP org, considering that integration into NGen is progressing.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+[project]
+name = "ngen-forcings"
+authors = [
+    { name = "Jason Ducker", email = "jason.ducker@noaa.gov" }
+]
+description = "NextGen Forcings Engine"
+readme = "README.md"
```
Comment (aaraney) [inline pyproject.toml:7]: Just to be on the safe side, I would recommend specifying a context type here. That will help software that consume wheel and source distribution archives property display the readme.
```suggestion
readme = { file = "README.md", content-type = "text/markdown" }
```

Comment (aaraney) [reply pyproject.toml:7]: While you are at it, I would also add the license too (goes in `project` section).

```
license = { file = "LICENSE" }
```

Comment (program--) [reply pyproject.toml]: Awesome, thank you! Added in 8d23251

**Review by jduckerOWP — APPROVED** [review]
All the changes look great to me Justin. Thanks again for your work on integrating the NextGen Forcings Engine.

**Pull request merged**